app/flatpak-cli-transaction.c

```diff
@@ -248,7 +248,7 @@ progress_changed_cb (FlatpakTransactionProgress *progress,
 {
   FlatpakCliTransaction *cli = data;
   FlatpakTransaction *self = FLATPAK_TRANSACTION (cli);
-  FlatpakTransactionOperation *op = flatpak_transaction_get_current_operation (self);
+  g_autoptr(FlatpakTransactionOperation) op = flatpak_transaction_get_current_operation (self);
```

Comment:
I didn't find `G_DEFINE_AUTOPTR_CLEANUP_FUNC` defined for `FlatpakTransactionOperation` in the codebase.
You need to define:
`G_DEFINE_AUTOPTR_CLEANUP_FUNC (FlatpakTransactionOperation, g_object_unref);` in flatpak-transaction.h header file I think.

Reply:
```
common/flatpak-transaction.h:89:G_DECLARE_FINAL_TYPE (FlatpakTransactionOperation, flatpak_transaction_operation, FLATPAK, TRANSACTION_OPERATION, GObject)
```

[`G_DECLARE_FINAL_TYPE`](https://developer.gnome.org/gobject/stable/gobject-Type-Information.html#G-DECLARE-FINAL-TYPE:CAPS):

> This results in the following things happening:
> […]
> * g_autoptr() support being added for your type, based on the type of your parent class